src/site/content/en/blog/virtualkeyboard/index.md

```diff
@@ -0,0 +1,186 @@
+if ('virtualKeyboard' in navigator) {
+  // The VirtualKeyboard API is supported!
+}
+```
+
+The VirtualKeyboard API is implemented as of Chromium&nbsp;94 on desktop and mobile.
+
+## Using the VirtualKeyboard API
+
+The VirtualKeyboard API adds a new interface `VirtualKeyboard` to the `navigator` object.
+
+### Opting in to the new virtual keyboard behavior
+
+To tell the browser that you are taking care of virtual keyboard occlusions yourself, you need to
+first opt in to the new virtual keyboard behavior by setting the boolean property `overlaysContent`
+to `true`.
+
+```js
+navigator.virtualKeyboard.overlaysContent = true;
+```
+
+### Showing and hiding the virtual keyboard
+
+You can programmatically show the virtual keyboard by calling its `show()` method. For this to work,
+the focused element needs to be a form control (such as a `textarea` element), or be an editing host
+(for example, by using the
+[`contenteditable`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/contentEditable)
+attribute). The method returns `undefined` but triggers a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.show();
+```
+
+To hide the virtual keyboard, call the `hide()` method. The method returns `undefined` but triggers
+a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.hide();
+```
+
+### Being informed of geometry changes
+
+Whenever the virtual keyboard appears or disappears, the `geometrychanged` event is dispatched. The
+event's `target` property contains the new geometry of the virtual keyboard inset as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect).
+
+```js
+navigator.virtualKeyboard.addEventListener('geometrychanged', (event) => {
+  const { x, y, width, height } = event.target;
+  console.log('Virtual keyboard geometry changed:', x, y, width, height);
+});
+```
+
+### Getting the current geometry
+
+You can get the current geometry of the virtual keyboard by looking at the `boundingRect` property.
+It exposes the current dimensions of the virtual keyboard as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect) object.
+
+```js
+const { x, y, width, height } = navigator.virtualKeyboard.boundingRect;
+console.log('Virtual keyboard geometry:', x, y, width, height);
+```
+
+### The CSS environment variables
+
+The VirtualKeyboard API exposes a set of CSS environment variables that provide information about
+the virtual keyboard's appearance.
+
+- `keyboard-inset-top`
+- `keyboard-inset-right`
+- `keyboard-inset-bottom`
+- `keyboard-inset-left`
+- `keyboard-inset-width`
+- `keyboard-inset-height`
+
+The virtual keyboard insets are six environment variables that define a rectangle by its top, right,
+bottom, and left insets from the edge of the viewport. The width and height insets are calculated
+from the remaining insets for developer ergonomics. The default value of each keyboard inset is
+`0px` if a fallback value is not provided, else it gets updated when the `boundingRect` value
+changes.
```

Comment:
```suggestion
changes.

You would typically use the environment variables as in the example below:

```css
.some-class {
  /**
   * Use a margin that corresponds to the virtual keyboard's height
   * if the virtual keyboard is shown, else use the fallback value of `50px`.
  margin-block-end: env(keyboard-inset-height, 50px);
}

.some-other-class {
  /**
   * Use a margin that corresponds to the virtual keyboard's height
   * if the virtual keyboard is shown, else use the default fallback value of `0px`.
  margin-block-end: env(keyboard-inset-height);
}
```
```